spoon/database/database.php

```diff
@@ -1029,6 +1052,17 @@ private function setPort($port)
 
 
 	/**
+	 * Set unix socket
+	 *
+	 * @param	string $unixSocket	The path to the unix socket (f.e.: used on MAC with MAMP).
+	 */
+	private function setUnixSocket($unixSocket)
+	{
+		$this->unixSocket = (string) $unixSocket;
```

Comment:
you cast this to string here, but it could also be null, your null check in line 154 can never work using this cast